runtime/lua/vim/_editor.lua

```diff
@@ -948,6 +948,63 @@ function vim._init_defaults()
   vim._init_default_autocmds()
 end
 
+table.pack = table.pack or function(...) return { n = select("#", ...), ... } end
```

Comment:
For what it's worth, there is `pack_len` and `unpack_len` in `vim.F` module that does this

Reply:
Should we have a `compat.lua` module where we set `table.pack` (to match Lua 5.2)? Assuming it's "less worse" to use a common Lua 5.2 name instead of a custom name?

Reply:
💯 that's two steps closer to killing `F.lua` -- leaves only `vim.F.npcall` as actually used in our codebase, that could just as well be moved directly under `vim`  to `_editor.lua`.

Reply:
But note that LuaJIT may be compiled with 5.2 extensions, in which case it already has `table.{un}pack`, so that would have to be checked first.

If we do go that route, we could also (conditionally) add useful extensions from openResty's LuaJIT2 like `table.isempty`, `table.isarray`, `table.nkeys`, and `table.clone` (which means we could deprecate `vim.tbl_isempty`).

Reply:
> If we do go that route, we could also (conditionally) add useful extensions from openResty's LuaJIT2 like table.isempty, table.isarray, table.nkeys, and table.clone (which means we could deprecate vim.tbl_isempty etc., or at least offer LuaJIT2 users access to the compiled version).

Sounds appealing. Would this play nicely with luals?